app/screens/settings/display_theme/theme_tiles.tsx

```diff
@@ -0,0 +1,165 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {useCallback, useMemo} from 'react';
+import {Text, TouchableOpacity, useWindowDimensions, View} from 'react-native';
+
+import CompassIcon from '@components/compass_icon';
+import {Preferences} from '@constants';
+import {useTheme} from '@context/theme';
+import {useIsTablet} from '@hooks/device';
+import {changeOpacity, makeStyleSheetFromTheme} from '@utils/theme';
+import {typography} from '@utils/typography';
+
+import ThemeThumbnail from './theme_thumbnail';
+
+const TILE_PADDING = 8;
+
+const getStyleSheet = makeStyleSheetFromTheme((theme) => {
+    return {
+        container: {
+            flexDirection: 'column',
+            padding: TILE_PADDING,
+            marginTop: 8,
+        },
+        imageWrapper: {
+            position: 'relative',
+            alignItems: 'flex-start',
+            marginBottom: 12,
+        },
+        thumbnail: {
+            resizeMode: 'stretch',
+        },
+        check: {
+            position: 'absolute',
+            right: 5,
+            bottom: 5,
+            color: theme.sidebarTextActiveBorder,
+        },
+        label: {
+            color: theme.centerChannelColor,
+            ...typography('Body', 200),
+        },
+        tilesContainer: {
+            marginBottom: 30,
+            paddingLeft: 8,
+            flexDirection: 'row',
+            flexWrap: 'wrap',
+            backgroundColor: theme.centerChannelBg,
+            borderTopWidth: 1,
+            borderBottomWidth: 1,
+            borderTopColor: changeOpacity(theme.centerChannelColor, 0.1),
+            borderBottomColor: changeOpacity(theme.centerChannelColor, 0.1),
+        },
+    };
+});
+
+type ThemeTileProps = {
+    action: (v: string) => void;
+    actionValue: string;
+    activeTheme: Theme;
+    isLandscape: boolean;
+    isTablet: boolean;
+    label: React.ReactElement;
+    selected: boolean;
+    tileTheme: Theme;
+};
+export const ThemeTile = ({
+    action,
+    actionValue,
+    activeTheme,
+    isLandscape,
+    isTablet,
+    label,
+    selected,
+    tileTheme,
+}: ThemeTileProps) => {
+    const style = getStyleSheet(activeTheme);
+    const {width: deviceWidth} = useWindowDimensions();
+
+    const layoutStyle = useMemo(() => {
+        const tilesPerLine = isLandscape || isTablet ? 4 : 2;
+        const fullWidth = isLandscape ? deviceWidth - 40 : deviceWidth;
```

Comment:
isLandscape as only applied to tablets? I just don't want to see code brought from v1 without reasoning

Reply:
corrected